server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -117,6 +117,20 @@ protected enum SyncGlobalCheckpointAfterOperation {
         AttemptAfterSuccess
     }
 
+    /**
+     * Execution of the replica action
+     */
+    protected enum ReplicaActionExecution {
+        /**
+         * Will bypass queue length and indexing pressure checks
+         */
+        SubjectToCircuitBreaker,
+        /**
+         * Will bypass queue length, indexing pressure and circuit breaker checks
+         */
+        BypassCircuitBreaker
```

Comment:
I think I'd prefer not to mention queue length and indexing pressure checks in these docs, since this enum doesn't affect the behaviour in this area. Eventually I think we'll generalise `PrimaryActionExecution` to apply to all transport actions, not just the primary phase of replication actions, and then we might expose that as an option here too.

Reply:
Yep, there's probably only a handful of primary/replica option combinations which make sense. Easily representable as a single enum with some analysis I imagine.